kubernetes/resource_kubernetes_service_account.go

```diff
@@ -284,6 +311,17 @@ func resourceKubernetesServiceAccountRead(ctx context.Context, d *schema.Resourc
 	}
 
 	defaultSecretName := d.Get("default_secret_name").(string)
+	b, err := serverVersionGreaterThanOrEqual(conn, "1.24.0")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if b {
+		diagMessages = append(diagMessages, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "'default_secret_name' is no longer applicable for Kubernetes 'v1.24.0' and above",
+			Detail:   "Starting from version 1.24.0 Kubernetes does not automatically generate a token for service accounts, in this case, `default_secret_name` will be empty",
```

Comment:
I think we usually we `"double quote"` attributes we reference in diagnostic messages rather than single quote them.

Reply:
Changed to double quote, thanks!